File: Sorting and Searching/Movie Festival/movieFestival.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.Scanner;

public class movieFestival {
    private static void solve(){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[][] timings = new int[n][2];
        for(int i=0;i<n;i++){
            timings[i][0] = scanner.nextInt();
            timings[i][1] = scanner.nextInt();
        }
        Arrays.sort(timings, new Comparator<int[]>() {
            public int compare(int[] a, int[] b){
                return Integer.compare(a[1], b[1]);
            }
        });
        int idx = 1, lastIdx = 0, ans = 1;
        while (idx < n) {
            if(timings[idx][0] >= timings[lastIdx][1]){
                ans++;
                lastIdx = idx;
            }
            idx++;
        }
        System.out.println(ans);
    }
    public static void main(String args[]){
        solve();
    }    
}

File: Sorting and Searching/Movie Festival/movie_festival.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int


void solve(){
	int n, ans = 1, temp;
	cin>>n;
	vector<vector<int>> timing(n, vector<int>(2));

	for(int i=0;i<n;i++){
		cin>>temp;
		timing[i][0] = temp;
		cin>>temp;
		timing[i][1] = temp;
		swap(timing[i][0], timing[i][1]);
	}
	sort(timing.begin(), timing.end());

	int idx = 1, lastIdx = 0;
	while(idx < n){
		if(timing[idx][1] >= timing[lastIdx][0]){
			ans++;
			lastIdx = idx;
		}
		idx++;
	}
	cout<<ans;
    return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}